src/integration-test/java/com/mongodb/jdbc/integration/testharness/IntegrationTestUtils.java

```diff
@@ -0,0 +1,481 @@
+            case "getBestRowIdentifier":
+                return executeGetBestRowIdentifier(metadataFunction, databaseMetaData);
+            case "getTables":
+                return executeGetTables(metadataFunction, databaseMetaData);
+            case "getUDTs":
+                return executeGetUDTs(metadataFunction, databaseMetaData);
+            default:
+                return executeDBMetadataHelper(functionName, metadataFunction, databaseMetaData);
+        }
+    }
+
+    // Can process methods that have String only arguments or no arguments
+    private ResultSet executeDBMetadataHelper(
+            String functionName, List<Object> functionArgs, DatabaseMetaData databaseMetaData)
+            throws SQLException, InvocationTargetException, IllegalAccessException {
+        Method[] m = DatabaseMetaData.class.getMethods();
+        for (Method method : m) {
+            if (method.getName().equals(functionName)) {
+                Class<?>[] types = method.getParameterTypes();
+                // Verifying all String types
+                for (Class<?> type : types) {
+                    if (type != String.class) {
+                        throw new IllegalArgumentException(
+                                "expected: "
+                                        + String.class.getName()
+                                        + " found: "
+                                        + type.getName());
+                    }
+                }
+                if (method.getReturnType() != ResultSet.class) {
+                    throw new IllegalArgumentException(
+                            "expected: "
+                                    + ResultSet.class.getName()
+                                    + " found: "
+                                    + method.getReturnType());
+                }
+                if (method.getParameterCount() == 0) {
+                    return (ResultSet) method.invoke(databaseMetaData);
+
+                } else {
+                    List<String> stringFunctionArgs =
+                            functionArgs
+                                    .stream()
+                                    .map(object -> Objects.toString(object, null))
+                                    .collect(Collectors.toList());
+
+                    return (ResultSet)
+                            method.invoke(
+                                    databaseMetaData,
+                                    (Object[]) stringFunctionArgs.toArray(new String[0]));
+                }
+            }
+        }
+        throw new IllegalArgumentException("function '" + functionName + "' not found");
+    }
+
+    private ResultSet executeGetIndexInfo(
+            List<Object> functionArgs, DatabaseMetaData databaseMetaData) throws SQLException {
+        int expectedColumnCount = 5;
+        if (functionArgs.size() != expectedColumnCount) {
+            throw new IllegalArgumentException(
```

Comment:
Throwing IllegalArgumentException here and in other parts of the code when the configuration test files are incorrect.  Used instead of assert as I think assert is better used for verification errors of the results.

Reply:
From standup, team suggested looking into implementing dynamic tests for this:
https://www.baeldung.com/junit5-dynamic-tests

Reply:
Made changes to run tests as DynamicTest.  With the change, it no longer supports the BeforeEach/AfterEach methods which I was using for creating and closing the connection.  I also removed the pattern matching I was using as each test will be run as separate tests.  The test description will be used for the test name.